mmdeploy/codebase/mmrotate/deploy/rotated_detection.py

```diff
@@ -0,0 +1,318 @@
+
+    Args:
+        model_cfg (mmcv.Config): The model config.
+        imgs (Sequence[str] | Sequence[np.ndarray]): Input image(s), accepted
+            data type are List[str], List[np.ndarray].
+        input_shape (list[int]): A list of two integer in (width, height)
+            format specifying input shape. Default: None.
+
+    Returns:
+        mmcv.Config: the model config after processing.
+    """
+    if model_cfg.data.test['type'] == 'ConcatDataset':
+        model_cfg.data.test.pipeline = \
+            model_cfg.data.test['datasets'][0].pipeline
+
+    is_ndarray = isinstance(imgs[0], np.ndarray)
+
+    if is_ndarray:
+        model_cfg.data.test.pipeline[0].type = 'LoadImageFromNdarray'
+
+    test_pipeline = model_cfg.data.test.pipeline
+    test_pipeline = replace_ImageToTensor(test_pipeline)
+    # for static exporting
+    if input_shape is not None:
+        test_pipeline[1].img_scale = tuple(input_shape)
+        test_pipeline[1].transforms[0].keep_ratio = False
+        test_pipeline[1].transforms[0].img_scale = tuple(input_shape)
+    model_cfg.data.test.pipeline = test_pipeline
+    return model_cfg
+
+
+@MMROTATE_TASK.register_module(Task.ROTATED_DETECTION.value)
+class RotatedDetection(BaseTask):
+    """Rotated detection task class.
+
+    Args:
+        model_cfg (mmcv.Config): Loaded model Config object..
+        deploy_cfg (mmcv.Config): Loaded deployment Config object.
+        device (str): A string represents device type.
+    """
+
+    def __init__(self, model_cfg: mmcv.Config, deploy_cfg: mmcv.Config,
+                 device: str):
+        super(RotatedDetection, self).__init__(model_cfg, deploy_cfg, device)
+
+    def init_backend_model(self,
+                           model_files: Optional[str] = None,
+                           **kwargs) -> torch.nn.Module:
+        """Initialize backend model.
+
+        Args:
+            model_files (Sequence[str]): Input model files.
+
+        Returns:
+            nn.Module: An initialized backend model.
+        """
+        from .rotated_detection_model import build_rotated_detection_model
+        model = build_rotated_detection_model(
+            model_files, self.model_cfg, self.deploy_cfg, device=self.device)
+        return model.eval()
+
+    def init_pytorch_model(self,
+                           model_checkpoint: Optional[str] = None,
+                           cfg_options: Optional[Dict] = None,
+                           **kwargs) -> torch.nn.Module:
+        """Initialize torch model.
+
+        Args:
+            model_checkpoint (str): The checkpoint file of torch model,
+                defaults to `None`.
+            cfg_options (dict): Optional config key-pair parameters.
+
+        Returns:
+            nn.Module: An initialized torch model generated by OpenMMLab
+                codebases.
+        """
+        from mmrotate.apis import init_detector
+        model = init_detector(self.model_cfg, model_checkpoint, self.device,
```

Comment:
There is no `init_detector` in `mmrotate.apis`.

Reply:
Yes, maybe we should add init_detector in `mmrotate.apis`. I'll try if there's another way.